javalib/src/main/scala/java/util/ArrayDeque.scala

```diff
@@ -397,10 +388,8 @@ class ArrayDeque[E] private (initialCapacity: Int)
       pollLast()
       true
     } else if (target < endIndex) {
-      // Shift elements from endIndex towards target
-      for (i <- target until endIndex - 1)
-        inner(i) = inner(i + 1)
-      inner(endIndex - 1) = null.asInstanceOf[E] // free reference for GC
+      System.arraycopy(inner, target + 1, inner, target, endIndex - target)
```

Comment:
This doesn't look right. Shouldn't it be
```suggestion
      System.arraycopy(inner, target + 1, inner, target, endIndex - (target + 1))
```